workflows/reference_files/2.4/reference_files.smk

```diff
@@ -383,6 +383,160 @@ rule get_mutect2_small_exac:
         """)
 
 
+### FOR HANDLING CAPTURE SPACE/TARGETED SEQUENCING DATA ###
+# Added by Chris
+# What this *should* do (if I have written these rules correctly) is obtain a capture space BED file
+# check the contig names against the reference (and replace them as necessary), pad, sort,merge, and generate a bgzip/tabix
+# indexed version of the BED as well as an interval list.
+
+def _check_capspace_provider(w):
+    # Checks to determine if both the capture space BED and associated reference genome
+    # are chr prefixed or not
+
+    # If this is specified as the "default", make sure we obtain the relevent capture space
+    default_key = "default-" + w.genome_build
+    if default_key not in config["capture_space"] and w.capture_space == default_key:
+        # aka if the user hasn't explicitly specified a default using a default name
+        capture_space = _get_default_capspace(w)
+    else:
+        capture_space = w.capture_space
+
+    genome_provider = config["genome_builds"][w.genome_build]["provider"]
+    bed_provider = config["capture_space"][capture_space]["provider"]
+    
+    # If the providers match (i.e. they share the same prefix), just use the downloaded version
+    if genome_provider == bed_provider:
+        return {'bed': expand(rules.download_capspace_bed.output.capture_bed, capture_space=capture_space, genome_build=w.genome_build)}
+    else:
+        # Prompt the prefix to be converted
+        chr_status = "chr" if genome_provider == "ucsc" else "no_chr"
+        return {'bed': expand(rules.add_remove_chr_prefix_bed.output.converted_bed, capture_space = capture_space, genome_build = w.genome_build, chr_status= chr_status)}
+
+
+def get_capture_space(sample_id, genome_build, seq_type, return_ext):
+    """
+    Obtain the corresponding capture space file (gz, interval list etc) for the specified sample
+    """
+
+    # Get the corresponding capture space for a given sample
+    sample_table = CFG["samples"]
+    sample = sample_table.loc[(sample_table['sample_id'] == sample_id) &
+            (sample_table['genome_build'] == genome_build) &
+            (sample_table['seq_type'] == seq_type)]
+    if len(sample) != 1:
+        raise AssertionError("Found %s matches when examining the sample table for pair \'%s\' \'%s\' \'%s\'" % (len(sample), sample_id, genome_build, seq_type))
+    panel = sample.iloc[0]['capture_space']
+
+    # If this panel is "none" (aka not specified) use the default for this reference genome
+    if panel == "none":
```

Comment:
Do you think it will be helpful to allow for more flexibility here, allowing for N/As, NAs, None, etc.. ?

Reply:
Yes, I'll have to do a bit more testing. Let me look into this